packages/react/src/useAction.ts

```diff
@@ -47,7 +47,11 @@ export const useAction = <
   action: F,
   options?: LimitToKnownKeys<Options, F["optionsType"]>
 ): ActionHookResult<
-  GadgetRecord<Select<Exclude<F["schemaType"], null | undefined>, DefaultSelection<F["selectionType"], Options, F["defaultSelection"]>>>,
+  F["hasReturnType"] extends true
+    ? any
+    : GadgetRecord<
+        Select<Exclude<F["schemaType"], null | undefined>, DefaultSelection<F["selectionType"], Options, F["defaultSelection"]>>
+      >,
```

Comment:
in my 🎩 i am currently seeing `Promise<any>` here, maybe it's a version bump problem